File: Sqrilizz-Reports/src/main/java/dev/sqrilizz/SQRILIZZREPORTS/TelegramManager.java
package dev.sqrilizz.SQRILIZZREPORTS;

import com.google.gson.JsonObject;
import okhttp3.*;
import java.io.IOException;
import java.util.concurrent.TimeUnit;

/**
 * Optimized Telegram Manager using lightweight HTTP client instead of heavy Telegram Bot API
 * This reduces JAR size by ~15-20MB
 */
public class TelegramManager {
    private static String botToken;
    private static String chatId;
    private static boolean enabled = false;
    private static OkHttpClient httpClient;
    
    private static final String TELEGRAM_API_URL = "https://api.telegram.org/bot";
    private static final MediaType JSON = MediaType.get("application/json; charset=utf-8");

    public static void initialize() {
        try {
            if (Main.getInstance().getConfig().getBoolean("telegram.enabled", false)) {
                botToken = Main.getInstance().getConfig().getString("telegram.token", "");
                chatId = Main.getInstance().getConfig().getString("telegram.chat-id", "");
                
                if (!botToken.isEmpty() && !chatId.isEmpty()) {
                    // Create lightweight HTTP client
                    httpClient = new OkHttpClient.Builder()
                        .connectTimeout(10, TimeUnit.SECONDS)
                        .writeTimeout(10, TimeUnit.SECONDS)
                        .readTimeout(30, TimeUnit.SECONDS)
                        .build();
                    
                    enabled = true;
                    Main.getInstance().getLogger().info("Optimized Telegram manager initialized successfully");
                } else {
                    Main.getInstance().getLogger().warning("Telegram bot not configured properly");
                }
            }
        } catch (Exception e) {
            Main.getInstance().getLogger().severe("Failed to initialize optimized Telegram manager: " + e.getMessage());
        }
    }

    public static boolean isEnabled() {
        return enabled && httpClient != null;
    }

    public static void sendReport(ReportManager.Report report) {
        if (!isEnabled()) return;

        Main.runTaskAsync(() -> {
            try {
                String message = String.format(
                    "*Новая жалоба*\n\n" +
                    "*От:* %s\n" +
                    "*На:* %s\n" +
                    "*Причина:* %s\n" +
                    "*Время:* %s\n" +
                    "*Координаты жалобщика:* %s\n" +
                    "*Координаты цели:* %s",
                    report.reporter,
                    report.target,
                    report.reason,
                    report.getFormattedTime(),
                    report.reporterLocation,
                    report.targetLocation
                );

                sendMessage(message);
            } catch (Exception e) {
                Main.getInstance().getLogger().warning("Failed to send Telegram report: " + e.getMessage());
            }
        });
    }

    public static void sendBugReport(ReportManager.Report report, String category) {
        if (!isEnabled()) return;

        Main.runTaskAsync(() -> {
            try {
                String message = String.format(
                    "*Новый баг-репорт*\n\n" +
                    "*Отправитель:* %s\n" +
                    "*Категория:* %s\n" +
                    "*Описание:* %s\n" +
                    "*Время:* %s\n" +
                    "*Координаты отправителя:* %s",
                    report.reporter,
                    category.toUpperCase(),
                    report.reason,
                    report.getFormattedTime(),
                    report.reporterLocation
                );

                sendMessage(message);
            } catch (Exception e) {
                Main.getInstance().getLogger().warning("Failed to send Telegram bug report: " + e.getMessage());
            }
        });
    }

    public static void sendStatusUpdate(
        ReportManager.Report report,
        String resolver,
        ReportManager.Status status
    ) {
        if (!isEnabled()) return;

        Main.runTaskAsync(() -> {
            try {
                String reportType = "BUG_REPORT".equals(report.target)
                    ? "баг-репорт"
                    : "жалоба";
                String message = String.format(
                    "*Статус репорта изменён*\n\n" +
                    "*ID:* %d\n" +
                    "*Тип:* %s\n" +
                    "*Отправитель:* %s\n" +
                    "*Цель:* %s\n" +
                    "*Статус:* %s\n" +
                    "*Модератор:* %s",
                    report.id,
                    reportType,
                    report.isAnonymous ? "Аноним" : report.reporter,
                    "BUG_REPORT".equals(report.target) ? "Сервер" : report.target,
                    getStatusLabel(status),
                    resolver
                );
                sendMessage(message);
            } catch (Exception e) {
                Main.getInstance().getLogger().warning("Failed to send Telegram status update: " + e.getMessage());
            }
        });
    }

    public static void sendMessage(String text) {
        if (!isEnabled()) return;

        try {
            JsonObject payload = new JsonObject();
            payload.addProperty("chat_id", chatId);
            payload.addProperty("text", text);
            payload.addProperty("parse_mode", "Markdown");

            RequestBody body = RequestBody.create(payload.toString(), JSON);
            Request request = new Request.Builder()
                .url(TELEGRAM_API_URL + botToken + "/sendMessage")
                .post(body)
                .build();

            try (Response response = httpClient.newCall(request).execute()) {
                if (!response.isSuccessful()) {
                    Main.getInstance().getLogger().warning("Failed to send Telegram message: " + response.code());
                }
            }
        } catch (IOException e) {
            Main.getInstance().getLogger().warning("Error sending Telegram message: " + e.getMessage());
        }
    }

    public static void shutdown() {
        if (httpClient != null) {
            httpClient.dispatcher().executorService().shutdown();
            httpClient.connectionPool().evictAll();
        }
        enabled = false;
    }

    // Utility method to test bot connection
    public static boolean testConnection() {
        if (!isEnabled()) return false;

        try {
            Request request = new Request.Builder()
                .url(TELEGRAM_API_URL + botToken + "/getMe")
                .build();

            try (Response response = httpClient.newCall(request).execute()) {
                return response.isSuccessful();
            }
        } catch (IOException e) {
            return false;
        }
    }

    private static String getStatusLabel(ReportManager.Status status) {
        return switch (status) {
            case IN_PROGRESS -> "На проверке";
            case RESOLVED -> "Решён";
            case NOT_A_BUG -> "Не баг";
            case NOT_A_VIOLATION -> "Не нарушение";
            case FALSE_REPORT -> "Ложный репорт";
            case CLOSED -> "Закрыт";
            default -> "Открыт";
        };
    }
}

File: Sqrilizz-Reports/src/main/kotlin/dev/sqrilizz/SQRILIZZREPORTS/TelegramManager.kt
package dev.sqrilizz.SQRILIZZREPORTS

import com.google.gson.JsonObject
import okhttp3.MediaType.Companion.toMediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody
import java.io.IOException
import java.util.concurrent.TimeUnit

object TelegramManager {

    private var botToken: String = ""
    private var chatId: String = ""
    private var enabled = false
    private var httpClient: OkHttpClient? = null

    private const val TELEGRAM_API_URL = "https://api.telegram.org/bot"
    private val JSON_MEDIA = "application/json; charset=utf-8".toMediaType()

    @JvmStatic
    fun initialize() {
        try {
            if (Main.getInstance().config.getBoolean("telegram.enabled", false)) {
                botToken = Main.getInstance().config.getString("telegram.token", "") ?: ""
                chatId = Main.getInstance().config.getString("telegram.chat_id", "") ?: ""

                if (botToken.isNotEmpty() && chatId.isNotEmpty()) {
                    httpClient = OkHttpClient.Builder()
                        .connectTimeout(10, TimeUnit.SECONDS)
                        .writeTimeout(10, TimeUnit.SECONDS)
                        .readTimeout(30, TimeUnit.SECONDS)
                        .build()

                    enabled = true
                    Main.getInstance().logger.info("Optimized Telegram manager initialized successfully")
                } else {
                    Main.getInstance().logger.warning("Telegram bot not configured properly")
                }
            }
        } catch (e: Exception) {
            Main.getInstance().logger.severe("Failed to initialize optimized Telegram manager: ${e.message}")
        }
    }

    @JvmStatic
    fun isEnabled(): Boolean = enabled && httpClient != null

    @JvmStatic
    fun sendReport(report: ReportManager.Report) {
        if (!isEnabled()) return

        Main.runTaskAsync {
            try {
                val message = buildString {
                    appendLine("\uD83D\uDEA8 *Новая жалоба*")
                    appendLine()
                    appendLine("*От:* ${report.reporter}")
                    appendLine("*На:* ${report.target}")
                    appendLine("*Причина:* ${report.reason}")
                    appendLine("*Время:* ${report.formattedTime}")
                    appendLine("*Координаты жалобщика:* ${report.reporterLocation}")
                    append("*Координаты цели:* ${report.targetLocation}")
                }
                sendMessage(message)
            } catch (e: Exception) {
                Main.getInstance().logger.warning("Failed to send Telegram report: ${e.message}")
            }
        }
    }

    @JvmStatic
    fun sendMessage(text: String) {
        if (!isEnabled()) return

        try {
            val payload = JsonObject().apply {
                addProperty("chat_id", chatId)
                addProperty("text", text)
                addProperty("parse_mode", "Markdown")
            }

            val body = payload.toString().toRequestBody(JSON_MEDIA)
            val request = Request.Builder()
                .url("$TELEGRAM_API_URL$botToken/sendMessage")
                .post(body)
                .build()

            httpClient?.newCall(request)?.execute()?.use { response ->
                if (!response.isSuccessful) {
                    Main.getInstance().logger.warning("Failed to send Telegram message: ${response.code}")
                }
            }
        } catch (e: IOException) {
            Main.getInstance().logger.warning("Error sending Telegram message: ${e.message}")
        }
    }

    @JvmStatic
    fun shutdown() {
        httpClient?.let {
            it.dispatcher.executorService.shutdown()
            it.connectionPool.evictAll()
        }
        enabled = false
    }

    @JvmStatic
    fun testConnection(): Boolean {
        if (!isEnabled()) return false

        return try {
            val request = Request.Builder()
                .url("$TELEGRAM_API_URL$botToken/getMe")
                .build()
            httpClient?.newCall(request)?.execute()?.use { it.isSuccessful } ?: false
        } catch (_: IOException) {
            false
        }
    }
}
